src/Security/Member_Validator.php

```diff
@@ -147,12 +147,8 @@ public function php($data)
             if (!isset($data['DirectGroups'])) {
                 $stillAdmin = false;
             } else {
-                $adminGroups = array_intersect(
-                    $data['DirectGroups'] ?? [],
-                    Permission::get_groups_by_permission('ADMIN')->column()
-                );
-
-                if (count($adminGroups ?? []) === 0) {
+                $adminGroups = Permission::get_groups_by_permission('ADMIN')->filter(['ID' => $data['DirectGroups']]);
+                if (!$adminGroups->exists()) {
```

Comment:
The optimisations here are:
1. Filter the query directly by `$data['DirectGroups']` - if there's no matches, the database doesn't have to grab and return results that we're just gonna filter out anyway.
2. Using `exists()` is more efficient than selecting, returning, and `array_intersect`ing some data.